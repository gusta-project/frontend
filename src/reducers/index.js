import { combineReducers } from 'redux';

import {
  reducer as application,
  initialState as applicationState
} from './application';
import { reducer as profile, initialState as profileState } from './profile';
import { reducer as toast, initialState as toastState } from './toast';

export const initialState = {
  application: applicationState,
  profile: profileState,
  toast: toastState
};

export default combineReducers({
  application,
  profile,
  toast
});
