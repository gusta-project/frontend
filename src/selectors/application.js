import dayjs from 'dayjs';
import { createSelector } from 'reselect';

export const getApplication = state => state.application;

export const isLoggingIn = createSelector(
  getApplication,
  application => Boolean(application.loggingIn)
);

export const isLoggingOut = createSelector(
  getApplication,
  application => Boolean(application.loggingOut)
);

export const getUser = createSelector(
  getApplication,
  application => application.user
);

export const getError = createSelector(
  getApplication,
  application => application.error
);

export const getAuthorization = createSelector(
  getApplication,
  application => application.authorization
);

export const getToken = createSelector(
  getAuthorization,
  authorization => authorization.accessToken
);

export const getTokenExpiration = createSelector(
  getAuthorization,
  authorization => authorization.expiration
);

export const isLoggedIn = createSelector(
  [getToken, getTokenExpiration],
  (token, expiration) => Boolean(token && dayjs().isBefore(dayjs(expiration)))
);

export const getRegistration = createSelector(
  getApplication,
  application => application.registration
);

export const isRegistering = createSelector(
  getRegistration,
  registration => registration.registering && !registration.complete
);
