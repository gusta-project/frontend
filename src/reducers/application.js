import { buildActions } from 'utils';

export const types = buildActions('application', [
  'INIT_APP',
  'LOGIN_USER',
  'REQUEST_TOKEN',
  'RECEIVE_TOKEN',
  'REQUEST_CURRENT_USER',
  'RECEIVE_CURRENT_USER',
  'LOGIN_USER_SUCCESS',
  'LOGIN_USER_FAILURE',
  'LOGOUT_USER',
  'LOGOUT_USER_SUCCESS',
  'LOGOUT_USER_FAILURE',
  'REGISTER_USER',
  'REGISTER_USER_SUCCESS',
  'REGISTER_USER_FAILURE'
]);

const initApp = () => ({
  type: types.INIT_APP
});

const loginUser = (emailAddress, password) => ({
  type: types.LOGIN_USER,
  emailAddress,
  password
});

const requestToken = (emailAddress, password) => ({
  type: types.REQUEST_TOKEN,
  emailAddress,
  password
});

const receiveToken = (token, expiration) => ({
  type: types.RECEIVE_TOKEN,
  expiration,
  token
});

const requestCurrentUser = () => ({
  type: types.REQUEST_CURRENT_USER
});

const receiveCurrentUser = user => ({
  type: types.RECEIVE_CURRENT_USER,
  user
});

const loginUserSuccess = user => ({
  type: types.LOGIN_USER_SUCCESS,
  user
});

const loginUserFailure = error => ({
  type: types.LOGIN_USER_FAILURE,
  error
});

const logoutUser = () => ({
  type: types.LOGOUT_USER
});

const logoutUserSuccess = () => ({
  type: types.LOGOUT_USER_SUCCESS
});

const logoutUserFailure = error => ({
  type: types.LOGOUT_USER_FAILURE,
  error
});

const registerUser = details => ({
  type: types.REGISTER_USER,
  details
});

const registerUserSuccess = () => ({
  type: types.REGISTER_USER_SUCCESS
});

const registerUserFailure = error => ({
  type: types.REGISTER_USER_FAILURE,
  error
});

export const actions = {
  initApp,
  loginUser,
  requestToken,
  receiveToken,
  requestCurrentUser,
  receiveCurrentUser,
  loginUserSuccess,
  loginUserFailure,
  logoutUser,
  logoutUserSuccess,
  logoutUserFailure,
  registerUser,
  registerUserSuccess,
  registerUserFailure
};

export const initialState = {
  loggingIn: false,
  loggingOut: false,
  user: null,
  error: null,
  authorization: {
    accessToken: null,
    refreshToken: null,
    expiration: null
  },
  registration: {
    registering: false,
    complete: false,
    error: null
  }
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.LOGIN_USER:
      return {
        ...state,
        loggingIn: true
      };
    case types.LOGOUT_USER:
      return {
        ...state,
        loggingOut: true
      };
    case types.REGISTER_USER:
      return {
        ...state,
        registration: {
          ...state.registration,
          registering: true,
          details: action.details
        }
      };
    case types.RECEIVE_TOKEN:
      return {
        ...state,
        authorization: {
          accessToken: action.token,
          expiration: action.expiration
        }
      };
    case types.RECEIVE_CURRENT_USER:
      return {
        ...state,
        user: action.user
      };
    case types.LOGOUT_USER_SUCCESS:
      return {
        ...state,
        loggingOut: false,
        user: null,
        authorization: initialState.authorization
      };
    case types.LOGIN_USER_SUCCESS:
      return {
        ...state,
        loggingIn: false
      };
    case types.LOGIN_USER_FAILURE:
    case types.LOGOUT_USER_FAILURE:
      return {
        ...state,
        loggingIn: false,
        loggingOut: false,
        error: action.error
      };
    case types.REGISTER_USER_SUCCESS:
      return {
        ...state,
        registration: {
          registering: false,
          complete: true,
          error: null
        }
      };
    case types.REGISTER_USER_FAILURE:
      return {
        ...state,
        registration: {
          registering: false,
          complete: true,
          error: action.error
        }
      };
    default:
      return state;
  }
};
