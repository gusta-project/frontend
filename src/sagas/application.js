import dayjs from 'dayjs';
import { all, call, put, takeLatest, take, select } from 'redux-saga/effects';

import request from 'utils/request';
import { actions, types } from 'reducers/application';
import { actions as toastActions } from 'reducers/toast';
import { isLoggedIn, getUser } from 'selectors/application';

// snake_cased variables here come from RFC 6749
/* eslint-disable camelcase */
function* requestTokenWorker({ emailAddress, password }) {
  try {
    const endpoint = {
      url: '/oauth/token',
      method: 'POST'
    };
    const data = {
      grant_type: 'password',
      username: emailAddress,
      password
    };
    const result = yield call(request.execute, { endpoint, data });

    // update token info in state or throw an error
    if (result.success) {
      const {
        data: {
          access_token: token,
          token_type: tokenType,
          expires_in: expiresIn
        }
      } = result.response;

      if (tokenType !== 'Bearer') {
        throw new Error(`Unable to use token of type ${tokenType}`);
      }

      const expiration = dayjs().add(expiresIn, 'seconds');

      yield put(actions.requestTokenSuccess({ token, expiration }));
    } else if (result.error) {
      throw result.error;
    } else {
      throw new Error('Request failed for an unspecified reason!');
    }
  } catch (error) {
    const { message } = error;

    yield put(actions.requestTokenFailure(error));
    yield put(
      toastActions.popToast({
        title: 'Error!',
        icon: 'times-circle',
        message
      })
    );
  }
}
/* eslint-enable camelcase */

function* requestCurrentUserWorker() {
  try {
    const endpoint = {
      url: '/user/current',
      method: 'GET'
    };
    const result = yield call(request.execute, { endpoint });

    // update user in state or throw an error
    if (result.success) {
      const {
        response: { data }
      } = result;

      yield put(actions.requestCurrentUserSuccess(data));
    } else if (result.error) {
      throw result.error;
    } else {
      throw new Error('Failed to get current user!');
    }
  } catch (error) {
    const { message } = error;

    yield put(actions.requestCurrentUserFailure(error));
    yield put(
      toastActions.popToast({
        title: 'Error',
        icon: 'times-circle',
        message
      })
    );
  }
}

function* loginUserWorker({ emailAddress, password }) {
  try {
    // first, check to see if we are already logged in
    const loggedIn = yield select(isLoggedIn);

    if (!loggedIn) {
      // obtain a bearer token
      // then, obtain current user information
      // use take because it is blocking
      yield put(actions.requestToken(emailAddress, password));
      const tokenResult = yield take([
        types.REQUEST_TOKEN_SUCCESS,
        types.REQUEST_TOKEN_FAILURE
      ]);

      if (tokenResult.type === types.REQUEST_TOKEN_FAILURE) {
        throw new Error('Failed to log in!');
      }

      const { token, expiration } = tokenResult;

      localStorage.setItem('accessToken', JSON.stringify(token));
      localStorage.setItem(
        'expiration',
        JSON.stringify(expiration.toISOString())
      );
    }

    yield put(
      toastActions.popToast({
        title: 'Logged in',
        icon: 'check',
        message: 'You have been authenticated.'
      })
    );

    let user = yield select(getUser);

    if (!user) {
      yield put(actions.requestCurrentUser());
      const currentUserResult = yield take([
        types.REQUEST_CURRENT_USER_SUCCESS,
        types.REQUEST_CURRENT_USER_FAILURE
      ]);

      if (currentUserResult.type === types.REQUEST_CURRENT_USER_FAILURE) {
        throw new Error('Failed to fetch current user!');
      }

      user = currentUserResult.user;

      if (!user) {
        throw new Error('Got invalid response to current user request!');
      }
    }

    yield put(actions.loginUserSuccess());
  } catch (error) {
    const { message } = error;

    yield put(actions.loginUserFailure(error));
    yield put(
      toastActions.popToast({
        title: 'Error',
        icon: 'times-circle',
        message
      })
    );
  }
}

function* logoutUserWorker() {
  try {
    // first, check to see if we are logged in
    const loggedIn = yield select(isLoggedIn);

    if (loggedIn) {
      localStorage.removeItem('accessToken');
      localStorage.removeItem('expiration');
    }

    yield put(
      toastActions.popToast({
        title: 'Logged out',
        icon: 'check',
        message: 'You have been logged out.'
      })
    );

    yield put(actions.logoutUserSuccess());
  } catch (error) {
    const { message } = error;

    yield put(actions.logoutUserFailure(error));
    yield put(
      toastActions.popToast({
        title: 'Error',
        icon: 'times-circle',
        message
      })
    );
  }
}

function* registerUserWorker({
  details: { emailAddress, password, username }
}) {
  try {
    const endpoint = {
      url: '/register',
      method: 'POST'
    };
    const data = {
      emailAddress,
      password,
      username
    };
    const result = yield call(request.execute, { endpoint, data });

    if (result.success) {
      yield put(actions.registerUserSuccess());
      yield put(
        toastActions.popToast({
          title: 'Success!',
          icon: 'check',
          message: 'Check your email for an activation link.'
        })
      );
    } else {
      throw result.error;
    }
  } catch (error) {
    const { message } = error;

    yield put(actions.registerUserFailure(error));
    yield put(
      toastActions.popToast({
        title: 'Error',
        icon: 'times-circle',
        message
      })
    );
  }
}

function* loginUserWatcher() {
  yield takeLatest(types.LOGIN_USER, loginUserWorker);
}

function* logoutUserWatcher() {
  yield takeLatest(types.LOGOUT_USER, logoutUserWorker);
}

function* requestTokenWatcher() {
  yield takeLatest(types.REQUEST_TOKEN, requestTokenWorker);
}

function* requestCurrentUserWatcher() {
  yield takeLatest(types.REQUEST_CURRENT_USER, requestCurrentUserWorker);
}

function* registerUserWatcher() {
  yield takeLatest(types.REGISTER_USER, registerUserWorker);
}

export const workers = {
  loginUserWorker,
  logoutUserWorker,
  registerUserWorker,
  requestTokenWorker,
  requestCurrentUserWorker
};

export const watchers = {
  loginUserWatcher,
  logoutUserWatcher,
  registerUserWatcher,
  requestTokenWatcher,
  requestCurrentUserWatcher
};

export default function* saga() {
  yield all(Object.values(watchers).map(watcher => watcher()));
}
