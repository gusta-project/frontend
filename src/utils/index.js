import dayjs from 'dayjs';

import { getApplication, getAuthorization } from 'selectors/application';

/* eslint-disable react/display-name, react/prop-types */

/**
 * Creates an object containing action constants namespaced under the specified reducer.
 *
 * @param {string} reducer The name of the reducer or module
 * @param {string[]} actions A list of action names
 * @return {object} Object with action name keys and full action string values
 */
export function buildActions(reducer, actions) {
  const result = {};

  for (const action of actions) {
    result[action] = `${reducer}/${action}`;
  }

  return result;
}

/**
 * Interpolate strings from the `params` object into the `url`.
 *
 * @param {object} endpoint Object with `url` and optional `params` object
 */
export const buildUrl = (endpoint) => {
  // this is defined by webpack.DefinePlugin
  const baseUrl = API_URL;
  const { url, params } = endpoint;

  if (!params) {
    return `${baseUrl}${url}`;
  }

  return Object.entries(params).reduce((resultUrl, entry) => {
    const [key, value] = entry;

    return resultUrl.replace(`{${key}}`, value);
  }, `${baseUrl}${url}`);
};

/**
 * Gets the initial Redux state from localStorage and the source code defaults.
 */
export const getInitialState = () => {
  // this needs to be require()d because an import results in a
  // circular dependency
  const { initialState } = require('reducers');

  try {
    const [accessToken, expiration] = [
      localStorage.getItem('accessToken'),
      localStorage.getItem('expiration')
    ];

    if (!accessToken) {
      return initialState;
    }

    const expirationDate = dayjs(JSON.parse(expiration));

    if (!expirationDate.isValid() || expirationDate.isBefore(dayjs())) {
      localStorage.removeItem('accessToken');
      localStorage.removeItem('expiration');
      return initialState;
    }

    return {
      ...initialState,
      application: {
        ...getApplication(initialState),
        authorization: {
          ...getAuthorization(initialState),
          accessToken: JSON.parse(accessToken),
          expiration: expirationDate
        }
      }
    };
  } catch {
    return initialState;
  }
};
