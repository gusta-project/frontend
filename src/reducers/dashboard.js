import { buildActions } from 'utils';

export const types = buildActions('dashboard', [
  'REQUEST_DASHBOARD',
  'REQUEST_DASHBOARD_SUCCESS',
  'SELECT_DASHBOARD',
  'SELECT_DASHBOARD_SUCCESS',
  'REQUEST_MIGRATIONS',
  'REQUEST_MIGRATIONS_SUCCESS',
  'REQUEST_MIGRATIONS_FAILURE',
  'REQUEST_STATS',
  'REQUEST_STATS_SUCCESS',
  'REQUEST_STATS_FAILURE'
]);

const requestDashboard = () => ({
  type: types.REQUEST_DASHBOARD
});

const requestDashboardSuccess = (dashboardComponent) => ({
  type: types.REQUEST_DASHBOARD_SUCCESS,
  dashboardComponent
});

const selectDashboard = ({ name, item }) => ({
  type: types.SELECT_DASHBOARD,
  name,
  item
});

const selectDashboardSuccess = (dashboardComponent) => ({
  type: types.SELECT_DASHBOARD_SUCCESS,
  dashboardComponent
});

const requestMigrations = () => ({
  type: types.REQUEST_MIGRATIONS
});

const requestMigrationsSuccess = (migrations) => ({
  type: types.REQUEST_MIGRATIONS_SUCCESS,
  migrations
});

const requestMigrationsFailure = (error) => ({
  type: types.REQUEST_MIGRATIONS_FAILURE,
  error
});

const requestStats = () => ({
  type: types.REQUEST_STATS
});

const requestStatsSuccess = (stats) => ({
  type: types.REQUEST_STATS_SUCCESS,
  stats
});

const requestStatsFailure = (error) => ({
  type: types.REQUEST_STATS_FAILURE,
  error
});

export const actions = {
  requestDashboard,
  requestDashboardSuccess,
  selectDashboard,
  selectDashboardSuccess,
  requestMigrations,
  requestMigrationsSuccess,
  requestMigrationsFailure,
  requestStats,
  requestStatsSuccess,
  requestStatsFailure
};

export const initialState = {
  dashboardComponent: {
    name: 'Home',
    item: null
  },
  migrations: [],
  stats: {}
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.REQUEST_DASHBOARD_SUCCESS:
      return {
        ...state,
        dashboardComponent: action.dashboardComponent
      };
    case types.SELECT_DASHBOARD_SUCCESS:
      return {
        ...state,
        dashboardComponent: action.dashboardComponent
      };
    case types.REQUEST_MIGRATIONS_SUCCESS:
      return {
        ...state,
        migrations: action.migrations
      };
    case types.REQUEST_MIGRATIONS_FAILURE:
      return {
        ...state,
        error: action.error
      };
    case types.REQUEST_STATS_SUCCESS:
      return {
        ...state,
        stats: action.stats
      };
    case types.REQUEST_STATS_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
