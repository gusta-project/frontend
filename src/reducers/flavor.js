import { buildActions } from 'utils';

export const types = buildActions('flavor', [
  'REQUEST_STASH',
  'REQUEST_STASH_SUCCESS',
  'REQUEST_STASH_FAILURE'
]);

const requestStash = () => ({
  type: types.REQUEST_STASH
});

const requestStashSuccess = stash => ({
  type: types.REQUEST_STASH_SUCCESS,
  stash
});

const requestStashFailure = error => ({
  type: types.REQUEST_STASH_FAILURE,
  error
});

export const actions = {
  requestStash,
  requestStashSuccess,
  requestStashFailure
};

export const initialState = {
  loaded: false,
  error: null,
  stash: []
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.REQUEST_STASH_SUCCESS:
      return {
        ...state,
        stash: action.stash,
        loaded: true
      };
    case types.REQUEST_STASH_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
