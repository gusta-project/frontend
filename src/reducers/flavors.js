import { buildActions } from 'utils';

export const types = buildActions('flavors', [
  'REQUEST_FLAVORS',
  'REQUEST_FLAVORS_SUCCESS',
  'REQUEST_FLAVORS_FAILURE'
]);

const requestFlavors = (pager) => ({
  type: types.REQUEST_FLAVORS,
  pager
});

const requestFlavorsSuccess = (flavors, pager) => ({
  type: types.REQUEST_FLAVORS_SUCCESS,
  flavors,
  pager
});

const requestFlavorsFailure = (error) => ({
  type: types.REQUEST_FLAVORS_FAILURE,
  error
});

export const actions = {
  requestFlavors,
  requestFlavorsSuccess,
  requestFlavorsFailure
};

export const initialState = {
  cache: [],
  collection: [],
  pager: {
    count: null,
    limit: 100,
    page: 1,
    pages: null
  }
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.REQUEST_FLAVORS_SUCCESS:
      return {
        ...state,
        cache: action.flavors,
        collection: action.flavors[action.pager.page],
        pager: {
          ...state.pager,
          ...action.pager
        }
      };
    case types.REQUEST_FLAVORS_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
