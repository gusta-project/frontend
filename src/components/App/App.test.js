import dayjs from 'dayjs';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import ConnectedApp, { App } from './App';
import { withMemoryRouter } from 'utils';
import { initialState } from 'reducers/application';

jest.mock('components/Header/Header', () =>
  require('utils').mockComponent('Header')
);
jest.mock('components/ToastDrawer/ToastDrawer', () =>
  require('utils').mockComponent('ToastDrawer')
);
jest.mock('pages/NotFound', () => require('utils').mockComponent('NotFound'));
jest.mock('pages/Home', () => require('utils').mockComponent('Home'));
jest.mock('pages/Login', () => require('utils').mockComponent('Login'));
jest.mock('pages/Register', () => require('utils').mockComponent('Register'));
jest.mock('pages/user/Profile', () =>
  require('utils').mockComponent('Profile')
);
jest.mock('pages/user/Recipes', () =>
  require('utils').mockComponent('Recipes')
);
jest.mock('pages/user/Favorites', () =>
  require('utils').mockComponent('Favorites')
);
jest.mock('pages/user/FlavorStash', () =>
  require('utils').mockComponent('FlavorStash')
);
jest.mock('pages/user/Settings', () =>
  require('utils').mockComponent('Settings')
);
jest.mock('pages/user/ShoppingList', () =>
  require('utils').mockComponent('ShoppingList')
);
jest.mock('pages/Calculator', () =>
  require('utils').mockComponent('Calculator')
);
jest.mock('pages/Flavors', () => require('utils').mockComponent('Flavors'));
jest.mock('pages/Recipe', () => require('utils').mockComponent('Recipe'));

describe('<App />', () => {
  const actions = {
    initApp: jest.fn()
  };
  const mockStore = configureStore();
  const store = mockStore({ application: initialState });
  const authedStore = mockStore({
    application: {
      ...initialState,
      authorization: {
        accessToken: '1234',
        expiration: dayjs().add(60, 'minutes')
      },
      user: {
        id: 123,
        name: 'Doe'
      }
    }
  });

  afterEach(() => {
    actions.initApp.mockReset();
  });

  it('calls initApp', () => {
    const RoutedApp = withMemoryRouter(App);

    renderer.create(
      <Provider store={store}>
        <RoutedApp actions={actions} />
      </Provider>
    );

    expect(actions.initApp).toHaveBeenCalled();
  });

  it('renders home by default', () => {
    const RoutedApp = withMemoryRouter(ConnectedApp);
    const component = renderer.create(
      <Provider store={store}>
        <RoutedApp />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders login page', () => {
    const RoutedApp = withMemoryRouter(ConnectedApp, {
      initialEntries: ['/login']
    });
    const component = renderer.create(
      <Provider store={store}>
        <RoutedApp />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders register page', () => {
    const RoutedApp = withMemoryRouter(ConnectedApp, {
      initialEntries: ['/register']
    });
    const component = renderer.create(
      <Provider store={store}>
        <RoutedApp />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders profile page', () => {
    const RoutedApp = withMemoryRouter(ConnectedApp, {
      initialEntries: ['/user/profile']
    });
    const component = renderer.create(
      <Provider store={authedStore}>
        <RoutedApp />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders user recipes page', () => {
    const RoutedApp = withMemoryRouter(ConnectedApp, {
      initialEntries: ['/user/recipes']
    });
    const component = renderer.create(
      <Provider store={authedStore}>
        <RoutedApp />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders favorites page', () => {
    const RoutedApp = withMemoryRouter(ConnectedApp, {
      initialEntries: ['/user/favorites']
    });
    const component = renderer.create(
      <Provider store={authedStore}>
        <RoutedApp />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders flavor stash page', () => {
    const RoutedApp = withMemoryRouter(ConnectedApp, {
      initialEntries: ['/user/flavor-stash']
    });
    const component = renderer.create(
      <Provider store={authedStore}>
        <RoutedApp />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders user settings page', () => {
    const RoutedApp = withMemoryRouter(ConnectedApp, {
      initialEntries: ['/user/settings']
    });
    const component = renderer.create(
      <Provider store={authedStore}>
        <RoutedApp />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders shopping list page', () => {
    const RoutedApp = withMemoryRouter(ConnectedApp, {
      initialEntries: ['/user/shopping-list']
    });
    const component = renderer.create(
      <Provider store={authedStore}>
        <RoutedApp />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders calculator page', () => {
    const RoutedApp = withMemoryRouter(ConnectedApp, {
      initialEntries: ['/calculator']
    });
    const component = renderer.create(
      <Provider store={store}>
        <RoutedApp />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders flavors page', () => {
    const RoutedApp = withMemoryRouter(ConnectedApp, {
      initialEntries: ['/flavors']
    });
    const component = renderer.create(
      <Provider store={store}>
        <RoutedApp />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders recipe page', () => {
    const RoutedApp = withMemoryRouter(ConnectedApp, {
      initialEntries: ['/recipe']
    });
    const component = renderer.create(
      <Provider store={store}>
        <RoutedApp />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('redirects to login for private route', () => {
    const RoutedApp = withMemoryRouter(ConnectedApp, {
      initialEntries: ['/user/recipes']
    });
    const component = renderer.create(
      <Provider store={store}>
        <RoutedApp />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
