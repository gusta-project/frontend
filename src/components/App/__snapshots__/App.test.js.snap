// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> redirects to login for private route 1`] = `
Array [
  <Header />,
  <ToastDrawer />,
  <Login />,
]
`;

exports[`<App /> renders calculator page 1`] = `
Array [
  <Header />,
  <ToastDrawer />,
  <Calculator />,
]
`;

exports[`<App /> renders favorites page 1`] = `
Array [
  <Header />,
  <ToastDrawer />,
  <Favorites />,
]
`;

exports[`<App /> renders flavor stash page 1`] = `
Array [
  <Header />,
  <ToastDrawer />,
  <FlavorStash />,
]
`;

exports[`<App /> renders flavors page 1`] = `
Array [
  <Header />,
  <ToastDrawer />,
  <Flavors />,
]
`;

exports[`<App /> renders home by default 1`] = `
Array [
  <Header />,
  <ToastDrawer />,
  <Home />,
]
`;

exports[`<App /> renders login page 1`] = `
Array [
  <Header />,
  <ToastDrawer />,
  <Login />,
]
`;

exports[`<App /> renders profile page 1`] = `
Array [
  <Header />,
  <ToastDrawer />,
  <Profile />,
]
`;

exports[`<App /> renders recipe page 1`] = `
Array [
  <Header />,
  <ToastDrawer />,
  <Recipe />,
]
`;

exports[`<App /> renders register page 1`] = `
Array [
  <Header />,
  <ToastDrawer />,
  <Register />,
]
`;

exports[`<App /> renders shopping list page 1`] = `
Array [
  <Header />,
  <ToastDrawer />,
  <ShoppingList />,
]
`;

exports[`<App /> renders user recipes page 1`] = `
Array [
  <Header />,
  <ToastDrawer />,
  <Recipes />,
]
`;

exports[`<App /> renders user settings page 1`] = `
Array [
  <Header />,
  <ToastDrawer />,
  <Settings />,
]
`;
